api/v1alpha1/resource_error.go

```diff
@@ -31,6 +35,22 @@ type ResourceError struct {
 	Recoverable bool `json:"recoverable"`
 }
 
+// SetResourceError fills in the ResourceError field in a resource with information from
+// an error. The ResourceError field is expected to be object.Status.Error
```

Comment:
Matt, have you tried doing this with generics?

Reply:
I looked into it, but I don't think it would make this any easier. @NateRoiger you're the generics expert now. Would this be better done with generics?

Reply:
The advantage in using generics is to enforce this at compile time (and to a lesser extent performance - which isn't of concern).

Reply:
One possible implementation.

```go
type resource[T any] interface {
	SetResourceError(err error)
}

func SetResourceError2[T any](object resource[T], err error) {
	object.SetResourceError(err)
}
```

```go
func (s *ClientMount) SetResourceError(err error) {
	s.Status.Error = NewResourceError("", err)
}
```

```go
dwsv1alpha1.SetResourceError2[*dwsv1alpha1.ClientMountStatus](clientMount, err)
```

I'd really like the status type to be inferred but I'm still learning how that might work.